### PR TITLE
React input field JS snippet

### PR DESCRIPTION
**Pull request by gcooney**
Adds a js snippet for interacting with a react input field.

### CODE REVIEW

File: mabl snippets/SetValueOnReactInputElement.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    const valueSetter = Object.getOwnPropertyDescriptor(element, 'value').set
+    const prototype = Object.getPrototypeOf(element)
+    const prototypeValueSetter = Object.getOwnPropertyDescriptor(prototype, 'value').set
+    if (valueSetter && valueSetter !== prototypeValueSetter) {
+      prototypeValueSetter.call(element, value)
+    } else {
+      valueSetter.call(element, value)
+    }
+  }
+  setNativeValue(textarea, '<value to input>')
```
Comment (liboz) [inline mabl snippets/SetValueOnReactInputElement.js]: this should be `textbox` right?

File: mabl snippets/SetValueOnReactInputElement.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    if (valueSetter && valueSetter !== prototypeValueSetter) {
+      prototypeValueSetter.call(element, value)
+    } else {
+      valueSetter.call(element, value)
+    }
+  }
+  setNativeValue(textbox, '<value to input>')
+    
+  var changeEvent = document.createEvent ('Event');
+  changeEvent.initEvent('change', true, false);
```
Comment (liboz) [inline mabl snippets/SetValueOnReactInputElement.js]: I see in the react testing library, they use `'input'`. Is there a reason to use a `'change'` event here instead? Do we need to potentially fire both?

File: mabl snippets/SetValueOnReactInputElement.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    const valueSetter = Object.getOwnPropertyDescriptor(element, 'value').set
+    const prototype = Object.getPrototypeOf(element)
+    const prototypeValueSetter = Object.getOwnPropertyDescriptor(prototype, 'value').set
+    if (valueSetter && valueSetter !== prototypeValueSetter) {
+      prototypeValueSetter.call(element, value)
+    } else {
+      valueSetter.call(element, value)
+    }
+  }
+  setNativeValue(textbox, '<value to input>')
```
Comment (liboz) [inline mabl snippets/SetValueOnReactInputElement.js]: Maybe it would be helpful to have both the value to input and the query be at the top as variables.

**Review by liboz — APPROVED** [review]
Just have a question about the event type. Otherwise looks good to me.

File: mabl snippets/SetValueOnReactInputElement.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
+"use strict";
+
+/**
+ * This javascript snippet sets a value on an input for applications using react.  mabl can generally support setting the input
+ * value directly by interacting with the page but this function provides a template for setting it via a javascript step, if needed
```
Comment (mwomer-mabl) [inline mabl snippets/SetValueOnReactInputElement.js:5]: This seems generic to me, and perhaps the check for the `getOwnPropertyDescriptor` bit could just be merged with our other click snippet?  Hadn't thought of that before.

Comment (gcooney) [reply mabl snippets/SetValueOnReactInputElement.js:5]: yeah, could be worth pulling into one generic thing.  To be fair, I didn't think of it either, just hunted down how react-testing library was simulating input events and found the linked react issue/comments.

**Pull request merged**